lib/Models/CatalogItem.js

```diff
@@ -984,7 +999,14 @@ function isEnabledChanged(catalogItem) {
             catalogItem._loadForEnablePromise = resolvedOrRejected ? undefined : loadPromise;
         }
 
-        catalogItem.isShown = true;
+        if (catalogItem.isShown) {
+            // If item already had the isShown flag but wasn't enabled, it wasn't actually shown, so call isShownChanged
+            // to update the view.
```

Comment:
Normally it would set `isEnabled = true`, except now `isEnabled` isn't set until the disclaimer is excepted. When it does get set, `isShown` is already true so the layer doesn't update and actually show itself.